### ISSUE TITLE
Next build hangs forever in linux and fails in CI

### ISSUE THREAD

**Issue by aak1247** [zeit/next.js#10954]
# Bug report

### In local machine

Build process can be finished in windows and wsl, but fails in linux(debian 5 and ubuntu 18.04 both fails). In local machine, it may hangs forever as:
> project@0.1.0 build /dir
> next build

> Using external babel configuration
> Location: "/dir/.babelrc"
Creating an optimized production build ...

and never finishs or throws error.

### In CI (gitlab-ci docker runner)

Build process seems to succeed but fails to export with this error:

> using build directory: /home/app/.next
{ Error: ENOENT: no such file or directory, open '/home/app/.next/BUILD_ID'
    at Object.openSync (fs.js:443:3)
    at readFileSync (fs.js:343:35)
    at _default (/home/app/node_modules/next/dist/export/index.js:1:3312)
    at nextExport (/home/app/node_modules/next/dist/cli/next-export.js:20:291)
    at commands.(anonymous function).then.exec (/home/app/node_modules/next/dist/bin/next:29:346)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/app/.next/BUILD_ID' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! meethive-pcwebsite@0.1.0 deploy: `next build && next export`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the meethive-pcwebsite@0.1.0 deploy script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-03-11T03_35_12_672Z-debug.log

## System information

- OS: windows 10(✔️ ), wsl arch linux(✔️ ), ubuntu 18.04(⛔️ ) and deepin 15(⛔️ ), ci(alpine and buster)
- Version of node: 10.15.1

## Additional context

I have tried to remove node_modules and use specified BUILD_DIR name and build id

package.json:
```json
{
  "name": "dir",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "export": "next export",
    "deploy": "next build && next export"
  },
  "dependencies": {
    "@zeit/next-less": "^1.0.1",
    "antd": "^3.25.2",
    "babel-plugin-import": "^1.13.0",
    "less": "^3.11.0",
    "less-loader": "^5.0.0",
    "less-vars-to-js": "^1.3.0",
    "mobx": "^5.15.0",
    "mobx-react": "^6.1.4",
    "next": "9.1.3",
    "null-loader": "^3.0.0",
    "react": "16.12.0",
    "react-dom": "16.12.0",
    "react-player": "^1.15.2"
  },
  "devDependencies": {
    "@babel/plugin-proposal-class-properties": "^7.7.0",
    "@babel/plugin-proposal-decorators": "^7.7.0"
  }

}
```

next.config.js
```js
const withLess = require('@zeit/next-less')
const lessToJS = require('less-vars-to-js')
const fs = require('fs')
const path = require('path')

// Where your antd-custom.less file lives
const themeVariables = lessToJS(
  fs.readFileSync(path.resolve(__dirname, './assets/antd-custom.less'), 'utf8')
)

module.exports = withLess({
  lessLoaderOptions: {
    javascriptEnabled: true,
    modifyVars: themeVariables // make your antd custom effective
  },
  webpack: (config, { isServer }) => {
    if (isServer) {
      const antStyles = /antd\/.*?\/style.*?/
      const origExternals = [...config.externals]
      config.externals = [
        (context, request, callback) => {
          if (request.match(antStyles)) return callback()
          if (typeof origExternals[0] === 'function') {
            origExternals[0](context, request, callback)
          } else {
            callback()
          }
        },
        ...(typeof origExternals[0] === 'function' ? [] : origExternals)
      ]

      config.module.rules.unshift({
        test: antStyles,
        use: 'null-loader'
      })
    }
    return config
  }
})


```

**Comment by timneutkens** [comment]
We'll need a complete reproduction in order to be able to investigate.

**Comment by aak1247** [comment]
✔️ Solved! It seems to be a case problem. A import path has wrong case( should start with lowercase but not), and the build process succeeds when change the start letter to lowercase.
Maybe we can produce a error when this happens?

**Comment by timneutkens** [comment]
We would still need a complete reproduction in order to investigate that

**Comment by Timer** [comment]
This behavior should be fixed in 9.3.5 or newer.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
